### DISCUSSION TITLE
How to add requirements to the docker image ?

### DISCUSSION THREAD

**Discussion by jkc20** [zauberzeug/nicegui#2929] [Q&A]
### Question

I took the nginx_subpath example and tried to run my own application. I got the error :ModuleNotFoundError: No module named 'passlib'
Usually one would put "RUN pip install -r requirements.txt" somewhere in the dockerfile. But I could not find the dockerfile for the pre-built multi-arch Docker image only for the dev-container. So the question is ether: How can I append my own dependencies to the pre-built image or where can I get the dockerfile of the pre-buit image to modify it ?

**Comment by rodja** [comment]
Instead of an pre-build image you can also specify a Dockerfile in the `docker-compose.yml`. We do this in our repository for one of our test-setups:

https://github.com/zauberzeug/nicegui/blob/d82241f876a4f327a5a1f03c5dbc84c80e8c402c/docker-compose.yml#L4-L6

**Comment by jkc20** [comment]
HI Rodja,
tx for your fast replay. I like to adapt the nginx_subpath example for my own app in production use. Sadly your pre-build image did not contain a dockerfile. Can I use the release.dockerfile from the nicegui repro to grap all nessesary dependencies ? Or is there some other documentation what is needed for production use ?

**Comment by rodja** [comment]
Simply write your own:

```dockerfile
FROM zauberzeug/nicegui:1.4.22

COPY requirements.txt requirements.txt
RUN pip install -r requirements.txt
```
